sched/clustermembership/dynamicCluster.go

```diff
@@ -1,49 +1,77 @@
 package cluster_membership
 
-/*
- * Represents a Cluster of nodes that can change over time.
- * Keeps a Map of the nodeId to Node for fast lookup of nodes.
- */
+//
+// Represents a Cluster of nodes that can change over time.
+// Keeps a Map of the nodeId to Node for fast lookup of nodes.
+//
 type dynamicCluster struct {
 	memberSet   map[string]bool
 	membersList []Node
+	updateCh    chan NodeUpdate
```

Comment:
You mean an Add & Remove Channel?

I decided against that because we want Adds & Removes to be in order that they are received.  Otherwise you run into problems with the state of a flapping node. Add Node, Remove Node, Add Node, etc... You have races and a node that is flapping for a while could get lost unless the updates are in order